marine/marine.py

```diff
@@ -236,3 +240,14 @@ def _collapse_macros(
             collapsed_result[field] = next(filter(None, possible_values), "")
 
         return collapsed_result
+
+    def _detect_encap(self, fields: List[str]) -> int:
+        fields_protocols = (field.split(".")[0].lower() for field in fields)
+        if [
```

Comment:
I think using `any` would be a more intuitive way of doing this, in terms of readability and performance

Reply:
Also, it would probably be faster to use `set` and the `intersection` method